### PR TITLE
Add block cache related DB properties

### PR DESCRIPTION
**Pull request by yiwu-arbug**
Summary:
Add DB properties "rocksdb.block-cache-capacity", "rocksdb.block-cache-usage", "rocksdb.block-cache-pinned-usage" to show block cache usage.

Test Plan:
See the new test.

### CODE REVIEW

File: db/internal_stats.cc [hunk 1]
```diff
@@ -830,6 +847,57 @@ bool InternalStats::HandleEstimateOldestKeyTime(uint64_t* value, DBImpl* /*db*/,
...
+  if (table_options->no_block_cache || block_cache == nullptr) {
+    return false;
+  }
+  *value = static_cast<uint64_t>(block_cache->GetUsage());
+  return true;
+}
+
+bool InternalStats::HandleBlockCachePinnedUsage(uint64_t* value, DBImpl* /*db*/,
+                                                Version* /*version*/) {
+  auto* table_factory = cfd_->ioptions()->table_factory;
```
Comment (maysamyabandeh) [inline db/internal_stats.cc]: minor: seems like we can factor out the initializations lines to a GetBlockCache private function.

File: db/db_properties_test.cc [hunk 2]
```diff
@@ -1436,6 +1436,98 @@ TEST_F(DBPropertiesTest, SstFilesSize) {
...
+      db_->GetIntProperty(DB::Properties::kBlockCachePinnedUsage, &value));
+  ASSERT_EQ(0, value);
+
+  // Insert unpinned item to the cache and check size.
+  constexpr size_t kSize1 = 50;
+  block_cache->Insert("item1", nullptr /*value*/, kSize1, nullptr /*deleter*/);
+  ASSERT_TRUE(db_->GetIntProperty(DB::Properties::kBlockCacheCapacity, &value));
+  ASSERT_EQ(kCapacity, value);
+  ASSERT_TRUE(db_->GetIntProperty(DB::Properties::kBlockCacheUsage, &value));
+  ASSERT_EQ(50, value);
```
Comment (maysamyabandeh) [inline db/db_properties_test.cc]: should 50 replace with kSize1?

File: db/db_properties_test.cc [hunk 3]
```diff
@@ -1436,6 +1436,98 @@ TEST_F(DBPropertiesTest, SstFilesSize) {
...
+  ASSERT_TRUE(db_->GetIntProperty(DB::Properties::kBlockCacheCapacity, &value));
+  ASSERT_EQ(kCapacity, value);
+  ASSERT_TRUE(db_->GetIntProperty(DB::Properties::kBlockCacheUsage, &value));
+  // Item 1 is evicted.
+  ASSERT_EQ(kSize2 + kSize3, value);
+  ASSERT_TRUE(
+      db_->GetIntProperty(DB::Properties::kBlockCachePinnedUsage, &value));
+  ASSERT_EQ(kSize2 + kSize3, value);
+
+  block_cache->Release(item2);
```
Comment (maysamyabandeh) [inline db/db_properties_test.cc]: would be interesting to check the size after the release as well.

**Comment by ajkr** [comment]
Just wondering, do we expose info about any other pointer options via DB properties? The reason I'm slightly concerned is because block cache, statistics, etc., are often shared across multiple DB instances, so a DB property like this can reflect info about other DB instances.

**Comment by yiwu-arbug** [comment]
@ajkr yes, that's true. But I think it is still good to have a single interface to expose stats, instead of requiring a lot of user code to obtain the stats directly from shared objects. What do you think?

**Pull request closed without merging**